src/runtime/PointerValue.h

```diff
@@ -608,6 +631,24 @@ class PointerValue : public gc {
         return (TemporalPlainDate*)this;
     }
 
+    TemporalZonedDateTime* asTemporalZonedDateTimeObject()
+    {
+        ASSERT(isTemporalZonedDateTimeObject());
+        return (TemporalZonedDateTime*)this;
```

Comment:
Just a thought: I see this is the same cast as above, but I think normally `reinterpret_cast` is used.